examples/usematches-loader-data/app/useOptionalUser.ts

```diff
@@ -13,8 +13,17 @@ function isUser(user: unknown): user is User {
   );
 }
 
-export default function useUser(): User | undefined {
-  const user = useLoaderStore("user");
+/**
+ * Build a custom hook for each data object
+ * of your loader data.
+ * Use useMatchesData to access loader data
+ * across your application.
+ * Use tiny-invariant and Typescript "is"
+ * to require data on runtime.
+ * Or return undefined if data is optional and not found.
+ */
+export default function useOptionalUser(): User | undefined {
+  const user = useMatchesData("root", "user");
```

Comment:
I would prefer:

```suggestion
  const { user } = useMatchesData("root");
```

Reply:
@kentcdodds Implemented! I had to change the return type of `useMatchesData` to safely destruct the user in `useOptionalUser`. Now the return value of `useMatchesData` cannot be undefined anymore but empty Json object. Lmk if that works!